console/console-init/ui/src/modules/device/components/AddDeviceWithJson/AddJsonUsingTemplate.tsx

```diff
@@ -0,0 +1,135 @@
+    enabled: true,
+    defaults: {
+      "content-type": "text/plain"
+    },
+    via: [],
+    ext: {}
+  }
+};
+interface IAddJsonUsingTemplate {
+  setDetail: (value: string) => void;
+  selectedTemplate: string;
+  setSelectedTemplate: (value: string) => void;
+}
+const AddJsonUsingTemplate: React.FunctionComponent<IAddJsonUsingTemplate> = ({
+  setDetail,
+  selectedTemplate,
+  setSelectedTemplate
+}) => {
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const onChange = (event: any) => {
+    const name: string = event.target.name;
+    if (name === "device-connected-directly") {
+      setSelectedTemplate(TemplateType.DIRECTLY_CONNECTED);
+    } else if (name === "device-via-gateway") {
+      setSelectedTemplate(TemplateType.VIA_GATEWAY);
+    }
+  };
+  const onClickTry = () => {
+    if (selectedTemplate === TemplateType.DIRECTLY_CONNECTED) {
+      setDetail(getFormattedJsonString(directlyConnectedDeviceTemplate));
+    } else if (selectedTemplate === TemplateType.VIA_GATEWAY) {
+      setDetail(getFormattedJsonString(connectedViaGatewayDeviceTemplate));
+    }
+  };
+  const onReaderLoad = (event: any) => {
+    try {
+      const data = JSON.parse(event.target.result);
+      setDetail(getFormattedJsonString(data));
+    } catch {
+      //TODO: Error handling for invalid json
+      console.log("Invalid Json");
+    }
+  };
+
+  const onClickUpload = () => {
+    setShowModal(true);
+  };
+
+  const onUploadFile = (value: string | File, filename: string, event: any) => {
+    var reader = new FileReader();
+    reader.onload = onReaderLoad;
+    reader.readAsText(event.target.files[0]);
+    setShowModal(false);
+  };
+
+  return (
+    <>
+      <Title headingLevel="h1" size="lg" id="title-select-json">
+        Select a JSON template for quick start
+      </Title>
+      <br />
+      <Radio
+        style={{ marginLeft: 10 }}
+        isLabelWrapped
+        isChecked={selectedTemplate === TemplateType.DIRECTLY_CONNECTED}
+        onClick={onChange}
+        label="Device connected directly"
+        id="radio-device-connected-directly"
+        name="device-connected-directly"
+      />
+      <br />
+      <Radio
+        style={{ marginLeft: 10 }}
+        isLabelWrapped
+        isChecked={selectedTemplate === TemplateType.VIA_GATEWAY}
+        onClick={onChange}
+        label="Device connected via gateway"
+        id="radio-device-via-gateway"
+        name="device-via-gateway"
+      />
+      <br />
+      <Button variant="secondary" onClick={onClickTry} id="btn-try-it">
+        Try it
+      </Button>
+      <br />
+      <br />
+      <Title headingLevel="h3" size="md" id="title-upload-json">
+        or Upload a JSON file
+      </Title>
+      <br />
+      <Button variant="secondary" onClick={onClickUpload} id="btn-upload-json">
+        <UploadIcon /> Upload a JSON file
+      </Button>
+      {showModal && (
+        <Modal title="" isOpen={showModal} onClose={() => setShowModal(false)}>
```

Comment:
I would like to suggest create a generic component for file upload and register that component in RootModal.tsx and use dispatch for open and close dialog/modal.